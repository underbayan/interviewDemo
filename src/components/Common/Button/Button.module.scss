@import "@demo/designSystem";

.buttonStyle {
  border: 1px solid var(--grays-5-color);
  color: var(--grays-3-color);
  box-sizing: content-box;

  &:disabled {
    color: var(--grays-5-color);
    background-color: none;
  }

  &:hover:disabled {
    cursor: not-allowed;
  }

  &:hover {
    color: var(--grays-5-color);
    border: 1px solid var(--grays-4-color);
    background-color: var(--grays-2-color);
    cursor: pointer;
  }
}