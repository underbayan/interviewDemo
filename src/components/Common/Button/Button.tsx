import { buttonStyle } from './Button.module.scss'
import { memo, MouseEventHandler } from "react"

interface ButtonProps {
  text: string;
  className?: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
}

export const Button = memo(({ text, className, onClick, disabled }: ButtonProps) => {
  return (
    <button disabled={disabled} className={`${buttonStyle} ${className}`} onClick={onClick}>
      {text}
    </button>
  )
})
