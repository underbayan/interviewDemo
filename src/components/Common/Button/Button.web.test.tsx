import { render } from '@testing-library/react';
import { test, expect } from 'vitest';
import { Button } from './Button';
test('Button renders correctly', () => {
  const { container } = render(<Button text="Click me"></Button>);
  expect(container).toMatchSnapshot();
});
