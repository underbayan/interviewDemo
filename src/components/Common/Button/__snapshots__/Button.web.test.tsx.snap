// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="undefined undefined"
  >
    Click me
  </button>
</div>
`;
